control/control.c

```diff
@@ -31,6 +33,7 @@ static void usage(int status)
 			"                           output also status of all components;\n"
 			"                           enabled components are enclosed in brackets\n"
 			"  -c, --ksmbd-version      output ksmbd version information and exit\n"
+			"  -v, --verbose            be more verbose\n"
```

Comment:
The expression "more verbose" is a bit weird. It's just verbose. Why not log-level? and you simply increased the log level value. Someone may be wondering if there is option to decrease log level. So I'd suggest explicitly setting it to PR_DEBUG. default log level in each utils is PR_INFO ? It might be meaningful if they are different. I did't understand why log_level is increased for verbose mode.

Reply:
@namjaejeon, I don't understand what your objection is. The phrase `be more verbose` is common. Each time `-v` is given, the utility should become more verbose. `ksmbd.adduser -a test -p '' -vvvv` *could* be more verbose than `ksmbd.adduser -a test -p '' -v`. Yes, currently `PR_DEBUG` is the only log level higher than the default `PR_INFO`.

Reply:
-vvvv will be meaningful if we have some option to decreasing log level. PR_INFO is log level by default for all utils. I am pointing out this. I hate having to add a check like the one below as -vv option that doesn't even work.

if (log_level < PR_DEBUG)

Reply:
@namjaejeon, how do you mean? Please, what program, that supports `--verbose`, also supports decreasing it?

Reply:
I mean -vv option doesn't work because there is no more higher log level than PR_DEBUG. So we should add unneeded check(if (log_level < PR_DEBUG)) for this. some option I said should not be named to "verbose" word. Probably will be "log-level". but we don't need to add such option in ksmbd-tools for now. So I am saying "more verbose" is weird.

See verbose option in filesystem utilities
```
Usage: fsck.ext4 [-panyrcdfktvDFV] [-b superblock] [-B blocksize]
		[-l|-L bad_blocks_file] [-C fd] [-j external_journal]
		[-E extended-options] [-z undo_file] device

Emergency help:
 -p                   Automatic repair (no questions)
 -n                   Make no changes to the filesystem
 -y                   Assume "yes" to all questions
 -c                   Check for bad blocks and add them to the badblock list
 -f                   Force checking even if filesystem is marked clean
 -v                   Be verbose
 -b superblock        Use alternative superblock
 -B blocksize         Force blocksize when looking for superblock
 -j external_journal  Set location of the external journal
 -l bad_blocks_file   Add to badblocks list
 -L bad_blocks_file   Set badblocks list
 -z undo_file         Create an undo file
```

```
Usage: fsck.exfat
	-r | --repair        Repair interactively
	-y | --repair-yes    Repair without ask
	-n | --repair-no     No repair
	-p | --repair-auto   Repair automatically
	-a                   Repair automatically
	-V | --version       Show version
	-v | --verbose       Print debug
	-h | --help          Show help
```

```
Usage: mkntfs [options] device [number-of-sectors]

Basic options:
    -f, --fast                      Perform a quick format(default)
    -Q, --quick                     Perform a quick format(default)
    -u, --full                      Perform a full format
    -L, --label STRING              Set the volume label
    -C, --enable-compression        Enable compression on the volume
    -I, --no-indexing               Disable indexing on the volume
    -n, --no-action                 Do not write to disk

Advanced options:
    -c, --cluster-size BYTES        Specify the cluster size for the volume
    -s, --sector-size BYTES         Specify the sector size for the device
    -p, --partition-start SECTOR    Specify the partition start sector
    -H, --heads NUM                 Specify the number of heads
    -S, --sectors-per-track NUM     Specify the number of sectors per track
    -z, --mft-zone-multiplier NUM   Set the MFT zone multiplier
    -T, --zero-time                 Fake the time to be 00:00 UTC, Jan 1, 1970
    -F, --force                     Force execution despite errors

Output options:
    -q, --quiet                     Quiet execution
    -v, --verbose                   Verbose execution
        --debug                     Very verbose execution

Help options:
    -V, --version                   Display version
    -l, --license                   Display licensing information
    -h, --help                      Display this help
```

Reply:
> some option I said should not be named to "verbose" word. Probably will be "log-level".

I canceled this mention.  I found such usage in other utilities. but still can not understand ++log_level although there is no more higher log level than PR_DEBUG.

Reply:
In case of fsck.ext4, just set verbose = 1.

Reply:
@namjaejeon, I can't make sense of this thread.

1. Why `--log-level`? `--verbose` is widespread. In what way log level and verbosity are not analogous?

2. If I do a builld of ksmbd-tools where default `log_level` is `PR_ERROR`, I want `-v` to increase verbosity. I don't want it to jump to the maximum log level. It is true that the default log level is PR_INFO, and PR_DEBUG is the only log level higher than it, but I don't see how that detracts from my point. And, regarding the phrase `be more verbose`, please grep your local filesystem or do a Google search for hundreds of examples. It does not imply that `-v -v` is any different to `-v`.

Please, help me understand your point of view.

Reply:
> If I do a builld of ksmbd-tools where default log_level is PR_ERROR, I want -v to increase verbosity.

Ah, Can you tell me where is setting log_level as PR_ERROR by default ? You can change to increase log_level with -v option and change word with "more verbose" when you change log_level to  PR_ERROR by default. e.g. In case of mke2fs in e2fsprogs utils, Since there is no level of verbose greater than 1, just set it to 1. and the repair code in e2fsprogs is treated as verbose++ because there is a log level when verbose is greater than 1.

misc/mke2fs.c
```
                case 'v':
                        verbose = 1;

```

misc/fsck.c
```
                        case 'V':
                                verbose++;
                                break;

        if (verbose > 1)
                printf(_("Finished with %s (exit status %d)\n"),
                       inst->device, inst->exit_status);

                        }

            if (verbose > 1)
                                printf("----------------------------------\n");
```

Reply:
And I think that "more" in "more verbose" mean increasing log level.

Reply:
`log_level` is `PR_INFO` by default. But if I happen to do a custom build for myself, I expect `-v` to increase logging incrementally. ksmbd-tools probably should not use `PR_ERROR` by default, since that would hide `pr_info()` messages.

Reply:
ksmbd-tools has 3 levels of verbosity. PR_ERROR means only pr_err() outputs. PR_INFO means both pr_err() and pr_info() output. PR_DEBUG means pr_err(), pr_info(), and pr_debug() output. This change adds 4th level of verbosity, PR_NONE, which hides all output.

Reply:
It is not appropriate to change the mainline for your custom build. First, if you have a usage that defaults to PR_ERROR, consider how to lower it without chnage the code in ksmbd-tools. I am not sure if it's appropriate, but if you look in samba's smb.conf, there is a log level parameter. If there is a way we can adjust the default log level in a similar way, I think that "more verbose" is reasonable.

Reply:
verbose mean showing debug level print. but we don't have only one level for verbose now. there is no more level than 1 from default level.

Reply:
Right, if you consider `log_level++` unacceptable compared to `log_level = PR_DEBUG`, then that's fine. I will remove `--verbose` from my changes. I will also remove the unimplemented `--verbose` stub that was in addshare and adduser. OK?

Reply:
> I will also remove the unimplemented --verbose stub that was in addshare and adduser. OK?

Ah, "Be more verbose" is not implemented, but "Be verbose" can be implemented with "log_level = PR_DEBUG" now. Am I missing something ?

Reply:
looks good to me about other patches except verbose in this PR.

Reply:
"be more verbose" and "be verbose" imply the same exact thing. "be more verbose" does not imply that "-v -v" is different to "-v" any more than the phrase "be verbose".

Reply:
If that's the author's intention, that's it. However, I am concerned about the part that can cause misunderstandings from users with various opinions and thoughts, and it would be nice if the seeds of misunderstanding could be cut. If you think that the two sentences are perfectly identical without misunderstanding from other users, it's okay to leave it as it is.

Reply:
This conversation is going nowhere. I'll make the changes you requested.

Reply:
Thanks a lot!